uvm/impl/com/untangle/uvm/AdminManagerImpl.java

```diff
@@ -196,14 +196,14 @@ public String getModificationState()
         if (blessedFile.exists() && blessedFile.lastModified() > zshHistoryFile.lastModified())
             return "blessed";
 
-        ExecManagerResult result = UvmContextImpl.context().execManager().exec("cat /root/.zsh_history | /usr/bin/wc -l");
+        ExecManagerResult result = UvmContextImpl.context().execManager().exec("cat /root/.zsh_history");
         int exitCode = result.getResult();
         String output = result.getOutput();
 
-        output = output.replaceAll("(\\r|\\n)", "");
+        int count = output.split("\r\n|\r|\n").length;
             
         if( exitCode == 0 ) {
-            return new String("yes (" + output + ")");
+            return new String("yes (" + count + ")");
```

Comment:
I detect that this code is problematic. According to the [Performance (PERFORMANCE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#performance-performance), [Dm: Method invokes inefficient new String(String) constructor (DM_STRING_CTOR)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#dm-method-invokes-inefficient-new-string-string-constructor-dm-string-ctor).

Using the `java.lang.String(String)` constructor wastes memory because the object so constructed will be functionally indistinguishable from the `String` passed as a parameter.  Just use the argument `String` directly.